indico/modules/events/registration/controllers/management/registration_tags.py

```diff
@@ -0,0 +1,115 @@
+from indico.modules.events.registration.models.tags import RegistrationTag
+from indico.modules.events.registration.views import WPManageRegistration
+from indico.util.i18n import _
+from indico.util.string import natural_sort_key
+from indico.web.args import use_kwargs
+from indico.web.flask.util import url_for
+from indico.web.util import jsonify_data, jsonify_template
+
+
+class RHManageRegistrationTags(RHManageRegFormsBase):
+    """List all registration tags for an event."""
+
+    def _process(self):
+        return WPManageRegistration.render_template('management/registration_tags.html', self.event)
+
+
+class RHRegistrationTagAdd(RHManageRegFormsBase):
+    """Add a new registration tag."""
+
+    def _process(self):
+        form = RegistrationTagForm(event=self.event)
+        if form.validate_on_submit():
+            tag = RegistrationTag(event=self.event)
+            form.populate_obj(tag, existing_only=True)
+            db.session.flush()
+            logger.info('Registration tag created by %s: %s', session.user, tag)
+            flash(_('A new tag has been created.'), 'success')
+            return jsonify_data(flash=False)
+
+        return jsonify_template('events/registration/management/registration_tag_edit.html',
+                                event=self.event, form=form, tag=None)
+
+
+class RHRegistrationTagDelete(RHManageRegFormsBase):
+    """Delete a registration tag."""
+
+    def _process(self):
+        tag = RegistrationTag.get_or_404(request.view_args['tag_id'])
+        db.session.delete(tag)
+        logger.info('Registration tag deleted by %s: %s', session.user, tag)
+        flash(_('Tag {} has been deleted.').format(tag.title), 'success')
+        return redirect(url_for('.manage_registration_tags', self.event))
+
+
+class RHRegistrationTagEdit(RHManageRegFormsBase):
+    """Modify an existing registration tag."""
+
+    def _process(self):
+        tag = RegistrationTag.get_or_404(request.view_args['tag_id'])
+        form = RegistrationTagForm(obj=tag)
+        if form.validate_on_submit():
+            form.populate_obj(tag, existing_only=True)
+            logger.info('Registration tag modified by %s: %s', session.user, tag)
+            flash(_('Tag {} has been modified.').format(tag.title), 'success')
+            return jsonify_data(flash=False)
+
+        return jsonify_template('events/registration/management/registration_tag_edit.html',
+                                event=self.event, form=form, tag=tag)
+
+
+def _assign_registration_tags(event, registrations, add, remove):
+    add_ids = [int(id) for id in add]
+    remove_ids = [int(id) for id in remove]
+    add_tags = RegistrationTag.query.with_parent(event).filter(RegistrationTag.id.in_(add_ids)).all()
+    remove_tags = RegistrationTag.query.with_parent(event).filter(RegistrationTag.id.in_(remove_ids)).all()
+
+    for reg in registrations:
+        reg.tags.update(add_tags)
+        reg.tags.difference_update(remove_tags)
```

Comment:
If you put `set()` around the two queries above, you can use the normal set operators here:

```
reg.tags |= add_tags
reg.tags -= remove_tags
```